File: iteration2/test/models/PrerequisiteTest.java
package iteration2.test.models;

import iteration2.src.enums.CourseResult;
import iteration2.src.models.Course;
import iteration2.src.models.CourseGrade;
import iteration2.src.models.Prerequisite;
import iteration2.src.models.Student;
import iteration2.src.models.Transcript;
import iteration2.src.utils.DatabaseManager;
import static org.junit.Assert.*;

import java.util.List;

import iteration2.src.enums.ApprovalStatus;
import iteration2.src.models.SelectedCourse;

public class PrerequisiteTest {

    @org.junit.Test
    public void checkPrequisiteCoursePassed() {

        Course course1 = new Course(5, 5, 1, "course1", "1", null, null, null);
        Course course4 = new Course(5, 5, 1, "course4", "4", null, null, null);
        
        List<Course> prerequisiteOfCourses2 = List.of(course1);
        Prerequisite prerequisite2 = new Prerequisite(prerequisiteOfCourses2);
        
        List<Course> prerequisiteOfCourses4 = List.of(course4);
        Prerequisite prerequisite4 = new Prerequisite(prerequisiteOfCourses4);
        
        Course course2 = new Course(10, 10, 1, "course2", "2", prerequisite2, null, null);
        Course course3 = new Course(7, 7 , 1, "course3", "3", prerequisite4, null, null);
        

        CourseGrade courseGrade1 = new CourseGrade(course1, "AA", CourseResult.PASSED);
        CourseGrade courseGrade2 = new CourseGrade(course2, "BB", CourseResult.PASSED);
        CourseGrade courseGrade3 = new CourseGrade(course3, "CC", CourseResult.PASSED);

        List<CourseGrade> courseGrades = List.of(courseGrade1, courseGrade2, courseGrade3);

        Transcript transcript = new Transcript(courseGrades);

        SelectedCourse selectedCourse1 = new SelectedCourse(course1, null);
        SelectedCourse selectedCourse2 = new SelectedCourse(course2, null);
        SelectedCourse selectedCourse3 = new SelectedCourse(course3, null);
        SelectedCourse selectedCourse4 = new SelectedCourse(course4, null);

        List<SelectedCourse> selectedCourses = List.of(selectedCourse1, selectedCourse2, selectedCourse3);

        Student student = new Student("1", "1", "eren", "duyuk", false, "a", "1", selectedCourses, null, null, transcript);

        
        assertTrue("Prerequisite should be passed", course2.checkPrerequisite(student));
        assertFalse("Prerequisite should not be passed", course3.checkPrerequisite(student));         

    }

}

File: iteration3/src/test/PrerequisiteTest.py
import unittest
from enums.CourseResult import CourseResult

from models.Course import Course
from models.CourseGrade import CourseGrade
from models.Prerequisite import Prerequisite
from models.SelectedCourse import SelectedCourse
from models.Student import Student
from models.Transcript import Transcript  

class PrerequisiteTest(unittest.TestCase):

    def testCheckPrequisiteCoursePassed(self):
        course1 = Course(5, 5, 1, "course1", "1", None, None, None)
        course4 = Course(5, 5, 1, "course4", "4", None, None, None)

        prerequisite_of_courses2 = [course1]
        prerequisite2 = Prerequisite(prerequisite_of_courses2)

        prerequisite_of_courses4 = [course4]
        prerequisite4 = Prerequisite(prerequisite_of_courses4)

        course2 = Course(10, 10, 1, "course2", "2", prerequisite2, None, None)
        course3 = Course(7, 7, 1, "course3", "3", prerequisite4, None, None)

        course_grade1 = CourseGrade(course1, "AA", CourseResult.PASSED)
        course_grade2 = CourseGrade(course2, "BB", CourseResult.PASSED)
        course_grade3 = CourseGrade(course3, "CC", CourseResult.PASSED)

        course_grades = [course_grade1, course_grade2, course_grade3]

        transcript = Transcript(course_grades)

        selected_course1 = SelectedCourse(course1, None)
        selected_course2 = SelectedCourse(course2, None)
        selected_course3 = SelectedCourse(course3, None)

        selected_courses = [selected_course1, selected_course2, selected_course3]

        student = Student("1", "1", "eren", "duyuk", False, "a", "1", 0, selected_courses, None, None, transcript)

        self.assertTrue(course2.checkPrerequisite(student), "Prerequisite should be passed")
        self.assertFalse(course3.checkPrerequisite(student), "Prerequisite should not be passed")

if __name__ == '__main__':
    unittest.main()
